pkg/recipes/terraform/install.go

```diff
@@ -78,5 +81,40 @@ func Install(ctx context.Context, installer *install.Installer, tfDir string) (s
 
 	logger.Info(fmt.Sprintf("Terraform latest version installed to: %q", execPath))
 
-	return execPath, nil
+	// Create a new instance of tfexec.Terraform with current Terraform installation path
+	tf, err := NewTerraform(ctx, tfDir, execPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Verify Terraform installation is complete before proceeding
+	for attempt := 0; attempt <= installRetryCount; attempt++ {
+		_, _, err = tf.Version(ctx, false)
+		if err != nil {
+			if attempt < installRetryCount {
+				logger.Info(fmt.Sprintf("Failed to verify Terraform installation: %s. Retrying after %d seconds", err.Error(), retryDelaySecs))
+				metrics.DefaultRecipeEngineMetrics.RecordTerraformInstallVerificationDuration(ctx, installStartTime,
+					[]attribute.KeyValue{
+						metrics.TerraformVersionAttrKey.String("latest"),
+						metrics.OperationStateAttrKey.String(metrics.FailedOperationState),
+					},
+				)
+				time.Sleep(time.Duration(retryDelaySecs) * time.Second)
+				continue
+			}
+			return nil, err
+		}
+	}
+
+	metrics.DefaultRecipeEngineMetrics.RecordTerraformInstallVerificationDuration(ctx, installStartTime,
+		[]attribute.KeyValue{
+			metrics.TerraformVersionAttrKey.String("latest"),
+			metrics.OperationStateAttrKey.String(metrics.SuccessfulOperationState),
+		},
+	)
+
+	// Configure Terraform logs once Terraform installation is complete
+	configureTerraformLogs(ctx, tf)
```

Comment:
Why was this moved here?

Reply:
This is where the original error was being thrown because in `configureTerraformLogs` we were using the TF executable to set logs. This is being separated out so we can first verify the TF installation completes and then continue on with configuring logs, downloading modules, etc